[Core] Use `Vector` for `MethodInfo::arguments`

AThousandShips opened the pull request:
One of the bigger offenders when it comes to using a bidirectional data type for random-access purposes, there aren't really any good reasons to use a `List` here that I can tell, we never do any insertions or erasures on it (one case in `Node` but it wasn't necessary, and IMO worsened readability)

Nothing should really change here except improving some data management, and locality, and making accessing the arguments easier, performance should be unaffected either way but haven't done any direct testing on that part, but I don't think this data is used in critical paths anyway

Follow-up to:
* https://github.com/godotengine/godot/pull/90705
<!--
Please target the `master` branch in priority.

Relevant fixes are cherry-picked for stable branches as needed by maintainers.

To speed up the contribution process and avoid CI errors, please set up pre-commit hooks locally:
https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html
-->

AThousandShips committed:
[Core] Use `Vector` for `MethodInfo::arguments`

KoBeWi commented on editor/connections_dialog.cpp:
```diff
@@ -282,10 +282,10 @@ List<MethodInfo> ConnectDialog::_filter_method_list(const List<MethodInfo> &p_me
 	List<MethodInfo> ret;
 
-	List<Pair<Variant::Type, StringName>> effective_args;
+	LocalVector<Pair<Variant::Type, StringName>> effective_args;
```
```suggestion
	LocalVector<Pair<Variant::Type, StringName>> effective_args;
	effective_args.reserve(p_signal.arguments.size() - unbind);
```
See https://github.com/godotengine/godot/pull/99936#issuecomment-2515610193

KoBeWi reviewed (approved):
Looks correct.

I wonder if it's worth it to switch to range iterators everywhere, i.e.
```C++
int i = 0;
for (const PropertyInfo &pi : method.arguments) {
	// stuff
	i++;
}
```
Avoids indexing the Vector, although it's cheap anyway.

AThousandShips commented:
Worth looking at yeah

RandomShaper commented on modules/gdscript/gdscript_utility_functions.cpp:
```diff
@@ -619,8 +619,8 @@ StringName GDScriptUtilityFunctions::get_function_return_class(const StringName
 	ERR_FAIL_NULL_V(info, Variant::NIL);
-	ERR_FAIL_COND_V(p_arg >= info->info.arguments.size(), Variant::NIL);
-	return info->info.arguments.get(p_arg).type;
+	ERR_FAIL_COND_V(p_arg >= (int)info->info.arguments.size(), Variant::NIL);
```
Would `ERR_FAIL_INDEX_V` be appropriate here?

AThousandShips replied:
Should be, will test!

AThousandShips commented:
Changed to using `ERR_FAIL_INDEX_V` and removed some of the typing changes that were added when this used `LocalVector` (due to signedness) and changed the iterations to use `int64_t` for correctness

Repiteo reviewed (approved):
Codestyle checks out. It'd be worth migrating to range iterators eventually, but that could always be tackled in a followup PR

AThousandShips commented:
I used for-each in several, most of the ones as far as I recall that don't use index because they need it

Repiteo commented:
Thanks!

AThousandShips commented:
Thank *you*!
